File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-13.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int length = nums.size();
        if (length < 2) {
            return length;
        }
        int l = 1;
        for (int r = 1; r < length; r++) {
            if (nums[r] != nums[r-1]) {
                nums[l] = nums[r];
                l++;
            }
        }
        return l;
    }
};

File: Data Structures & Algorithms/house-robber/submission-13.py
class Solution:
    def __init__(self):
        self.mp = {}
    def rob(self, nums: List[int]) -> int:
        if tuple(nums) in self.mp:
            return self.mp[tuple(nums)]
        if len(nums) <= 2:
            self.mp[tuple(nums)] = max(nums)
            return self.mp[tuple(nums)]
        self.mp[tuple(nums)] = max(nums[-1] + self.rob(nums[:-2]), self.rob(nums[:-1]))
        return self.mp[tuple(nums)]